src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -16,7 +17,7 @@ namespace Stratis.Bitcoin.Features.BlockStore
     /// <summary>
     /// <see cref="IBlockRepository"/> is the interface to all the logics interacting with the blocks stored in the database.
     /// </summary>
-    public interface IBlockRepository : IBlockStore
+    public interface IBlockRepository : IBlockStore, ITipProvider
```

Comment:
Does this really need to be on the interface? just keep it on the implementation no?

Reply:
aren't supposed all IBlockRepository implementation to handle the Tip?
If yes than it makes sense to have it at interface level